service/test/integrationdap_test.go

```diff
@@ -0,0 +1,120 @@
+package service_test
+
+import (
+	"bytes"
+	"net"
+	"testing"
+	"time"
+
+	protest "github.com/go-delve/delve/pkg/proc/test"
+	"github.com/go-delve/delve/service"
+	"github.com/go-delve/delve/service/dap"
+)
+
+func startDAPServer(t *testing.T) (server *dap.Server, addr string) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	server = dap.NewServer(&service.Config{
+		Listener:       listener,
+		Backend:        "default",
+		DisconnectChan: nil,
+	})
+	server.Run()
+	// Give server time to start listening for clients
+	time.Sleep(100 * time.Millisecond)
+	return server, listener.Addr().String()
+}
+
+func expectMessage(t *testing.T, client *DAPClient, want []byte) {
+	got, err := client.ReadBaseMessage()
+	if err != nil {
+		t.Error(err)
+	}
+	if !bytes.Equal(got, want) {
+		t.Errorf("\ngot  %q\nwant %q", got, want)
+	}
+}
+
+// name is for _fixtures/<name>.go
+func runTest(t *testing.T, name string, test func(c *DAPClient, f protest.Fixture)) {
+	var buildFlags protest.BuildFlags
+	fixture := protest.BuildFixture("helloworld", buildFlags)
+
+	server, addr := startDAPServer(t)
+	client := NewDAPClient(addr)
+	defer client.Close()
+	defer server.Stop()
+
+	test(client, fixture)
+}
+
+func TestStopOnEntry(t *testing.T) {
+	runTest(t, "helloworld", func(client *DAPClient, fixture protest.Fixture) {
+		client.InitializeRequest()
+		expectMessage(t, client, []byte(`{"seq":0,"type":"response","request_seq":0,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true}}`))
```

Comment:
I'm not sure about this method of asserting serialized bytes. It has some issues:
1) Order can change, and we'll have to rewrite a ton of tests
2) It's not clear what each expect is testing - I would expect some higher level primitives here

In general, the client should emulate an actual DAP client, e.g. events should update some state in it. This will be necessary to use Delve's integration tests anyway.

For now, I'd add a couple of higher level expect variants that test some things we need, but not necessarily all fields - comparisons to serialized `[]byte` should go away